lib/xchain/provider/fetch.go

```diff
@@ -433,3 +422,26 @@ func headTypeFromConfLevel(conf xchain.ConfLevel) (ethclient.HeadType, bool) {
 func reqToChainVersion(req xchain.ProviderRequest) chainVersion {
 	return chainVersion{ID: req.ChainID, ConfLevel: req.ConfLevel}
 }
+
+func getLogs(ctx context.Context, rpcClient ethclient.Client, contractAddr common.Address, blockHash common.Hash, topicName string) ([]types.Log, error) {
+	metadata, err := bindings.OmniPortalMetaData.GetAbi()
+	if err != nil {
+		return nil, errors.Wrap(err, "get abi")
+	}
+
+	topics, err := abi.MakeTopics([]any{metadata.Events[topicName].ID})
+	if err != nil {
+		return nil, errors.Wrap(err, "make topics")
+	}
+
+	logs, err := rpcClient.FilterLogs(ctx, ethereum.FilterQuery{
+		BlockHash: &blockHash,
+		Addresses: []common.Address{contractAddr},
+		Topics:    topics,
```

Comment:
note we could combine xmsgs and xreceipt into a single request and then parse logs based on topic. But maybe this isn't worth it...

Reply:
Yeah, doesn't seem super worth it to me.